hazelcast/src/main/java/com/hazelcast/nio/tcp/TcpIpConnectionManager.java

```diff
@@ -58,21 +58,18 @@
 import static com.hazelcast.internal.metrics.ProbeLevel.MANDATORY;
 import static com.hazelcast.internal.util.counters.MwCounter.newMwCounter;
 import static com.hazelcast.nio.IOUtil.closeResource;
+import static com.hazelcast.spi.properties.GroupProperty.BIND_SPOOFING_CHECKS;
 import static com.hazelcast.util.Preconditions.checkNotNull;
 import static com.hazelcast.util.ThreadUtil.createThreadPoolName;
 import static java.lang.Boolean.parseBoolean;
-import static java.lang.System.getProperty;
 
 public class TcpIpConnectionManager implements ConnectionManager, PacketHandler {
 
     private static final int RETRY_NUMBER = 5;
     private static final int DELAY_FACTOR = 100;
     private static final int SCHEDULER_POOL_SIZE = 4;
 
-    // TODO Introducing this to allow disabling the spoofing checks on-demand
-    // if there is a use-case that gets affected by the change. If there are no reports of misbehaviour we can remove than in
-    // next release.
-    private static final boolean SPOOFING_CHECKS = parseBoolean(getProperty("hazelcast.nio.tcp.spoofing.checks", "false"));
+    private static final boolean SPOOFING_CHECKS = parseBoolean(BIND_SPOOFING_CHECKS.getSystemProperty());
```

Comment:
I think that still parsing the system property defeats the purpose of having the property defined as `GroupProperty`:
- you can still only set at as system property, not as hazelcast property in declarative or programmatic config
- being a `static final` variable, you can only set it once per JVM, while in general we allow several Hazelcast instances per JVM and each instance can have its own set of `GroupProperty`'s so it defies the expected behavior of a `GroupProperty`.

If we are to move this into a `GroupProperty` I think this should be a non-static final field, populated in `TcpIpConnectionManager`'s constructor from a value resolved via `node.getProperties().getBoolean(GroupProperty.BIND_SPOOFING_CHECKS);`.

Reply:
very good suggestion. thx